src/Microsoft.NET.Sdk.Functions.MSBuild/Tasks/BuildFunctions.cs

```diff
@@ -29,8 +31,50 @@ public override bool Execute()
                 File.WriteAllText(hostJsonPath, hostJsonString);
             }
 
-            return FunctionJsonConvert.TryConvert(TargetPath, OutputPath, Log);
+            Assembly taskAssembly = typeof(BuildFunctions).GetTypeInfo().Assembly;
+#if NET46
```

Comment:
We need to change this condition. Anything under #if NET46 gets called when full framework msbuild is invoked. Full framework msbuild could be invoked for both apps targeting netfx & netstandard. We need to change this check to runtimeidentifier.

https://github.com/dotnet/sdk/blob/ac2507b9611def0ceb4af7ed2d2b479d82066bae/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.TargetFrameworkInference.targets#L57-L59
We can get the runtimeidentifier from msbuild and if app targets netstandard, then we need to call "dotnet console.dll" & if app target netfx, then we need to call "console.exe"

Reply:
how do you get the identifier from msbuild?

Reply:
You can just pass this additional property from these 2 targets - TargetFrameworkIdentifier 
https://github.com/Azure/azure-functions-vs-build-sdk/blob/master/src/Microsoft.NET.Sdk.Functions.MSBuild/Targets/netstandard1.0/Microsoft.NET.Sdk.Functions.Build.targets#L31
& 
https://github.com/Azure/azure-functions-vs-build-sdk/blob/master/src/Microsoft.NET.Sdk.Functions.MSBuild/Targets/netstandard1.0/Microsoft.NET.Sdk.Functions.Publish.targets#L142

and check the condition for '.NETFramework'

Reply:
See my comment below, this shouldn't be required. We should be fine running the exe on full framework builds and dotnet on core.